### PR TITLE
[11.0] [ADD] stock_picking_comment_template

### PR DESCRIPTION
**Pull request by emagdalenaC2i**
Adds comments on stock picking

### CODE REVIEW

**Comment by emagdalenaC2i** [comment]
@pedrobaeza @xavierjimenez and @cubells Please, could you take a look at this PR?

**Comment by emagdalenaC2i** [comment]
I'm not sure why it's red. Is it because 100% coverage is not achieved?

**Comment by pedrobaeza** [comment]
Yeah, this module lacks tests. They are not required, but recommended.

**Comment by emagdalenaC2i** [comment]
There are some test, but the coverage in this repository is 100%.
I will need some help to add more test and achive the 100%

**Comment by pedrobaeza** [comment]
Clicking on Details of codecov/project and clicking afterwards in "Diff" page, you will see in red the lines that are not being run because they are not tested nor executed on normal module install/update flow. This has nothing to do with the rest of the repository. Things you can test:

* The onchanges to have the expected results.
* Other methods following a flow in Odoo (the commercial partner thing is something I'm commenting now in the review).

File: stock_picking_comment_template/models/__init__.py [hunk 1]
```diff
@@ -0,0 +1,4 @@
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+from . import stock_picking
+from . import res_partner
```
Comment (pedrobaeza) [inline stock_picking_comment_template/models/__init__.py]: Add an extra trailing line

File: stock_picking_comment_template/models/res_partner.py [hunk 2]
```diff
@@ -0,0 +1,19 @@
+# Copyright 2019 C2i Change 2 improve - Eduardo Magdalena <emagdalena@c2i.es>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+from odoo import api, fields, models
+
+
+class ResPartner(models.Model):
+    _inherit = "res.partner"
+
+    comment_template_id = fields.Many2one(
```
Comment (pedrobaeza) [inline stock_picking_comment_template/models/res_partner.py]: This shouldn't go here in this module. There's a general module for that (`base_comment_template`)

File: stock_picking_comment_template/readme/CONTRIBUTORS.rst [hunk 3]
```diff
@@ -0,0 +1,5 @@
+* `C2i Change 2 improve <https://www.c2i.es>`_:
+
+  * Eduardo Magdalena <emagdalena@c2i.es>
+
+Do not contact contributors directly about support or help with technical issues.
```
Comment (pedrobaeza) [inline stock_picking_comment_template/readme/CONTRIBUTORS.rst:5]: This text is not propagated to the README.rst. Have you run the script? I think we can put this text on the master template for adding it in all templates.

**Review by pedrobaeza — CHANGES_REQUESTED** [review]

File: tests/test_stock_picking_report.py [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+            'name': 'Partner Test'
+        })
+        self.picking_model = self.env['stock.picking']
+        self.picking = self.picking_model.create({
+            'partner_id': self.partner.id,
+            'comment_template1_id': self.before_comment.id,
+            'comment_template2_id': self.after_comment.id
+        })
+
+        self.invoice._set_note1()
```
Comment (pedrobaeza) [inline tests/test_stock_picking_report.py]: Why testing here invoices? This is for pickings...

File: stock_picking_comment_template/readme/CONTRIBUTORS.rst (hunk 3 above)
Comment (emagdalenaC2i) [reply stock_picking_comment_template/readme/CONTRIBUTORS.rst:5]: No, I didn't run the script

File: stock_picking_comment_template/models/res_partner.py (hunk 2 above)
Comment (emagdalenaC2i) [reply stock_picking_comment_template/models/res_partner.py]: @pedrobaeza I did the same that in account_invoice_comment_template.
See 
https://github.com/OCA/account-invoice-reporting/blob/11.0/account_invoice_comment_template/models/res_partner.py

Sorry, I need a more detailed explanation

File: stock_picking_comment_template/readme/CONTRIBUTORS.rst (hunk 3 above)
Comment (pedrobaeza) [reply stock_picking_comment_template/readme/CONTRIBUTORS.rst:5]: Then please do it for properly updating this README.

File: stock_picking_comment_template/models/res_partner.py (hunk 2 above)
Comment (pedrobaeza) [reply stock_picking_comment_template/models/res_partner.py]: This is an incorrect split between modules (not your mistake, but something inherited). This should go on `base_comment_template`, and be removed from `account_invoice_comment_template`. Historically, only `account_invoice_comment_template` module existed, and then it was split in `base_...` and `account_...`. There, the mistake was to not move that code into base module.

File: stock_picking_comment_template/models/stock_picking.py [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+
+from odoo import api, fields, models
+
+
+class StockPicking(models.Model):
+    """Add text comment"""
+
+    _inherit = "stock.picking"
+
+    comment_template1_id = fields.Many2one('base.comment.template',
```
Comment (rousseldenis) [inline stock_picking_comment_template/models/stock_picking.py:12]: @emagdalenaC2i Should those fields be 'copy=False' ?

File: stock_picking_comment_template/models/stock_picking.py [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+                                           string='Top Comment Template')
+    comment_template2_id = fields.Many2one('base.comment.template',
+                                           string='Bottom Comment Template')
+    note1 = fields.Html('Top Comment')
+    note2 = fields.Html('Bottom Comment')
+
+    @api.onchange('comment_template1_id')
+    def _set_note1(self):
```
Comment (rousseldenis) [inline stock_picking_comment_template/models/stock_picking.py:20]: @emagdalenaC2i Functionally, is it good to erase a  note field that could have been modified by user (he maybe doesn't want to have his modifications lost)?

**Review by rousseldenis — COMMENTED** [review]
Some questions

File: stock_picking_comment_template/models/res_partner.py (hunk 2 above)
Comment (emagdalenaC2i) [reply stock_picking_comment_template/models/res_partner.py]: So, should I create another PR adding that code to (base_comment_template)?
or should I keep it as it is now, and then create a few PR to change (base_comment_template, account_invoice_comment_template and sale_comment_template) and this one?

File: stock_picking_comment_template/readme/CONTRIBUTORS.rst (hunk 3 above)
Comment (emagdalenaC2i) [reply stock_picking_comment_template/readme/CONTRIBUTORS.rst:5]: I tried, but I have installed the OCA Maintainers Tools and tried to login with (oca-github-login myusername) but it does nothing. And I can not run the script oca-gen-addon-readme

Comment (rousseldenis) [reply stock_picking_comment_template/readme/CONTRIBUTORS.rst:5]: You can locally install all maintainer-tools, executing setup from the repo directly. So, command line tools will be available.

File: stock_picking_comment_template/models/stock_picking.py (hunk 5 above)
Comment (emagdalenaC2i) [reply stock_picking_comment_template/models/stock_picking.py:12]: I don't think so. @pedrobaeza what do you think?

Comment (pedrobaeza) [reply stock_picking_comment_template/models/stock_picking.py:12]: Yeah, I think they should be preserved on duplication, as you might want to have the same comments. note1/note2 might be arguable to be preserved if they have custom changes though.

File: stock_picking_comment_template/readme/CONTRIBUTORS.rst (hunk 3 above)
Comment (emagdalenaC2i) [reply stock_picking_comment_template/readme/CONTRIBUTORS.rst:5]: ok, done

File: stock_picking_comment_template/models/res_partner.py (hunk 2 above)
Comment (emagdalenaC2i) [reply stock_picking_comment_template/models/res_partner.py]: @pedrobaeza I prefer to leave the base_comment_template module and this one for the moment as it is and create one or more PR later for account_invoice_comment_template and the rest of the modules

**Comment by emagdalenaC2i** [comment]
@pedrobaeza @xavierjimenez and @cubells Please, could you help me to increase the coverage of the tests? Thanks

**Comment by pedrobaeza** [comment]
The red parts are not tested:

seleccion_016

**Pull request closed without merging**